packages/node_modules/glimmer-syntax/lib/parser/handlebars-node-visitors.ts

```diff
@@ -107,7 +107,11 @@ export default {
       changeLines = leadingNewlineDifference(content.original, content.value);
     }
 
-    this.tokenizer.line = this.tokenizer.line + changeLines;
+    if (changeLines) {
```

Comment:
Based on https://github.com/wycats/handlebars.js/blob/9365b8290070f34bf797c836aed4335ce6a4094f/lib/handlebars/compiler/whitespace-control.js#L178-L194, it looks like `rightStripped` only indicates that some whitespace was stripped, and does not guarantee that we have `changeLines` of more than 1.